app/models/payment.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Payments made by an agency to a provide for a particular site
+class Payment < UuidApplicationRecord
+  belongs_to :agency
+  belongs_to :site
+
+  validates :amount, numericality: { greater_than: 0.00 }
+  validates :care_finished_on, presence: true # TODO: use Date validator
+  validates :care_started_on, presence: true # TODO: use Date validator
+  validates :discrepancy, numericality: true
+  validates :paid_on, presence: true # TODO: use Date validator
+
+  before_validation { |payment| payment.slug = generate_slug("#{payment.care_started_on}#{payment.site_id}#{payment.agency_id}") }
```

Comment:
Can we change this slug to `payment.site_id`, `payment.paid_on`, `payment.agency_id` and `payment.agency_amount` per comments from before?

Reply:
oops. missed that.  will do.